core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildStep.java

```diff
@@ -642,6 +664,14 @@ public NativeImageInvokerInfo build() {
                     enableHttpsUrlHandler = true;
                 }
 
+                assert nativeImageFeatures != null && !nativeImageFeatures.isEmpty()
```

Comment:
`assert`?

We don't really use it in the codebase

Reply:
Should I add an if-statement and through an exception instead?

Reply:
Yes please

Reply:
Done